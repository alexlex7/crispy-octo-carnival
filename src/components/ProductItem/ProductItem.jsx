import React from 'react';
import PropTypes from 'prop-types';
import styles from './ProductItem.module.css';
import ProductButton from '../ProductButton/ProductButton';
import { ReactComponent as FavoriteIcon } from '../../assets/images/heart.svg';

export default function ProductItem({ item, handleFavorite }) {
  const { id, name, price, imageURL, isFavorite } = item;

  const handleChange = (e) => {
    handleFavorite(id, e.target.checked);
  };

  return (
    <div className={styles.listItem}>
      <div className={styles.thumb}>
        <img src={imageURL} alt={name} title={name} />
      </div>
      <div className={styles.content}>
        <div className={styles.textWrapper}>
          <p className={styles.text}>{name}</p>
          <div className={styles.btnWrapper}>
            <label className={isFavorite ? styles.labelActive : styles.label}>
              <input
                type="checkbox"
                className={styles.checkbox}
                checked={isFavorite}
                onChange={handleChange}
              />
              <FavoriteIcon className={isFavorite ? styles.icon : ''} />
            </label>
            <ProductButton type="share" />
          </div>
        </div>
        <div className={styles.container}>
          <p className={styles.price}>${price}</p>
          <button className={styles.btn} type="button">
            Buy Now
          </button>
        </div>
      </div>
    </div>
  );
}

ProductItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    imageURL: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }),
  handleFavorite: PropTypes.func,
};
