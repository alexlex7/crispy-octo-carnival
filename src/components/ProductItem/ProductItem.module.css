.listItem {
  width: 380px;

  font-family: 'DM Sans';
  font-style: normal;
  font-weight: 500;

  background: #ffffff;
  box-shadow: 0px 13px 26px rgba(0, 0, 0, 0.07);
}
.thumb {
  width: 380px;
  height: 246px;
}

.thumb > img {
  display: block;
  width: 100%;
  height: auto;
  object-fit: contain;
}

.content {
  padding: 30px 24px 27px 24px;
}

.textWrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;

  margin-bottom: 21px;
}

.text {
  font-size: 20px;
  line-height: 27px;

  color: #242424;
}

.btnWrapper {
  display: flex;
  column-gap: 8px;
}

.container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.price {
  font-size: 24px;
  line-height: 31px;

  color: #154444;
}

.btn {
  padding: 6px 0px;
  width: 79px;

  font-size: 12px;
  line-height: 21px;
  color: #ffffff;

  background-color: #244d4d;
  border: none;
  transition: 0.3s;
}

.btnActive {
  background-color: #ef5423;
}

.activeIcon {
  fill: white;
}

.btn:hover,
.btn:focus {
  background-color: #3b7d7d;
}

.checkbox {
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.label {
  width: 26px;
  height: 26px;

  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  user-select: none;
  border-radius: 50%;

  background-color: #ecf4ff;
}

.label:hover {
  background-color: antiquewhite;
}

.labelActive {
  composes: label;
  background-color: #ef5423;
}

.labelActive:hover {
  background-color: #fa3f06;
}

.icon {
  fill: white;
}
