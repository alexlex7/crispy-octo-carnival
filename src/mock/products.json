[
  {
    "id": "1",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "2",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "3",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  },
  {
    "id": "4",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "5",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "6",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  },
  {
    "id": "7",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "8",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "9",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  }
]
