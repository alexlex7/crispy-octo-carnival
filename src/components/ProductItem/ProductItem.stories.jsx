import React from 'react';

import ProductItem from './ProductItem';
import products from '../../mock/products.json';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Products/ProductItem',
  component: ProductItem,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <ProductItem {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  item: products[0],
};
