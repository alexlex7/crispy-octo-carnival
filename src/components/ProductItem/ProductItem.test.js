import { render, screen } from '@testing-library/react';
import ProductItem from './ProductItem';
import products from '../../mock/products.json';

test('render product item', () => {
  const { asFragment } = render(<ProductItem item={products[0]} />);

  const text = screen.getByText(`${products[0].name}`);
  expect(text).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
