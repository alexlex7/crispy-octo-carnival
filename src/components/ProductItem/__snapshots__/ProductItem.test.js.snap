// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product item 1`] = `
<DocumentFragment>
  <div
    class="listItem"
  >
    <div
      class="thumb"
    >
      <img
        alt="White Swan Chair"
        src="http://localhost:8080/images/furniture/chair.jpg"
        title="White Swan Chair"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="textWrapper"
      >
        <p
          class="text"
        >
          White Swan Chair
        </p>
        <div
          class="btnWrapper"
        >
          <label
            class="label"
          >
            <input
              class="checkbox"
              type="checkbox"
            />
            <svg
              class=""
            >
              heart.svg
            </svg>
          </label>
          <button
            class="btn"
            type="button"
          >
            <svg
              class="icon"
            >
              share.svg
            </svg>
          </button>
        </div>
      </div>
      <div
        class="container"
      >
        <p
          class="price"
        >
          $40
        </p>
        <button
          class="btn"
          type="button"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
